sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2CProperties.java

```diff
@@ -84,6 +88,44 @@
 
     private String replyUrl = "{baseUrl}/login/oauth2/code/";
 
+    /**
+     * AAD B2C endpoint base uri.
+     */
+    @URL(message = "baseUri should be valid URL")
+    private String baseUri;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        if (StringUtils.isEmpty(tenant) && StringUtils.isEmpty(baseUri)) {
+            throw new AADB2CConfigurationException("'tenant' and 'baseUri' at least configure one item.");
+        }
+        // baseUri points to Global env by default
+        if (StringUtils.isEmpty(baseUri) && StringUtils.hasText(tenant)) {
+            baseUri = String.format("https://%s.b2clogin.com/%s.onmicrosoft.com/", tenant, tenant);
+        } else {
+            baseUri = addSlash(baseUri);
+        }
+    }
+
+    private String addSlash(@URL String uri) {
+        return uri.endsWith("/") ? uri : uri + "/";
+    }
+
+    /**
+     * Get tenant name according the tenant base uri endpoint.
+     * @return tenant name
+     */
+    public String getTenantName() {
+        if (StringUtils.hasText(tenant)) {
+            return tenant;
+        }
+        String[] uriParts = baseUri.split("/");
+        if (uriParts.length == 4) {
+            return uriParts[2].split("\\.")[0];
+        }
+        return null;
```

Comment:
not consistent with our description, if both tenant and base-uri set, only base-uri will take effect.